crates/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -1927,25 +1927,37 @@ fn quote_fqname_as_type_path(includes: &mut Includes, fqname: &str) -> TokenStre
     quote!(#expr)
 }
 
-fn collect_examples(docs: &Docs) -> Examples {
-    // TODO(#2919): `cpp` examples are not required for now, so just default to empty
-    Examples::collect(docs, "cpp", &["```cpp,ignore"], &["```"]).unwrap_or_default()
-}
+/* fn collect_examples(docs: &Docs) -> Examples {
+    Examples::collect(docs, "cpp", &["```cpp,ignore"], &["```"], false).unwrap_or_default()
+} */
 
 fn quote_docstrings(docs: &Docs) -> TokenStream {
     let mut lines = crate::codegen::get_documentation(docs, &["cpp", "c++"]);
 
-    let examples = collect_examples(docs);
+    // TODO(#2919): `cpp` examples are not required for now
+    let examples = collect_examples(docs, "cpp", false).unwrap_or_default();
     if !examples.is_empty() {
         lines.push(String::new());
-        let section_title = if examples.count == 1 {
+        let section_title = if examples.len() == 1 {
             "Example"
         } else {
             "Examples"
         };
         lines.push(format!("## {section_title}"));
         lines.push(String::new());
-        lines.extend(examples.lines.into_iter().map(|line| format!(" {line}")));
+        let mut examples = examples.into_iter().peekable();
+        while let Some(example) = examples.next() {
+            if let Some(title) = example.base.title {
+                lines.push(format!(" ### {title}"));
+            }
+            lines.push(" ```cpp,ignore".into());
+            lines.extend(example.content.iter().map(|line| format!(" {line}")));
+            lines.push(" ```".into());
+            if examples.peek().is_some() {
+                // blank line between examples
+                lines.push(" ".into());
```

Comment:
no need for trailing whitespace though 😬 

```suggestion
                lines.push("".into());
```